tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/PublicAssetMethodContract.scala

```diff
@@ -0,0 +1,659 @@
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.{BeforeEach, Test}
+import play.api.libs.json.{JsString, Json}
+
+import scala.jdk.CollectionConverters._
+
+object PublicAssetMethodContract {
+  case class UploadResponse(blobId: String, contentType: String, size: Long)
+}
+
+trait PublicAssetMethodContract {
+  @BeforeEach
+  def setUp(server: GuiceJamesServer): Unit = {
+    server.getProbe(classOf[DataProbeImpl])
+      .fluent()
+      .addDomain(DOMAIN.asString)
+      .addUser(BOB.asString(), BOB_PASSWORD)
+      .addUser(ANDRE.asString(), ANDRE_PASSWORD)
+
+    requestSpecification = baseRequestSpecBuilder(server)
+      .setAuth(authScheme(UserCredential(BOB, BOB_PASSWORD)))
+      .addHeader(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .build()
+      .log().ifValidationFails()
+  }
+
+  private def uploadAsset(content: Array[Byte] = "Your asset content here".getBytes,
+                          contentType: String = "image/png"): UploadResponse = {
+    val uploadResponse: String = `given`()
+      .basePath("")
+      .contentType(contentType)
+      .body(content)
+    .when
+      .post(s"/upload/$ACCOUNT_ID")
+    .`then`
+      .extract
+      .body
+      .asString
+
+    val blobId: String = (Json.parse(uploadResponse) \ "blobId").as[String]
+    val size: Long = (Json.parse(uploadResponse) \ "size").as[Long]
+    val assetType: String = org.apache.james.mailbox.model.ContentType.of((Json.parse(uploadResponse) \ "type").as[String])
+      .mimeType().asString()
+    UploadResponse(blobId, assetType, size)
+  }
+
+  @Test
+  def createValidRequestShouldReturnSuccessResponse(): Unit = {
+    val uploadResponse: UploadResponse = uploadAsset()
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:public:assets"],
+         |  "methodCalls": [
+         |    [
+         |      "PublicAsset/set", {
+         |        "accountId": "$ACCOUNT_ID",
+         |        "create": {
+         |          "4f29": {
+         |            "blobId": "${uploadResponse.blobId}"
+         |          }
+         |        }
+         |      }, "0"
+         |    ]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`()
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |    "PublicAsset/set",
+           |    {
+           |        "accountId": "$ACCOUNT_ID",
+           |        "oldState": "$${json-unit.ignore}",
+           |        "newState": "$${json-unit.ignore}",
+           |        "created": {
+           |            "4f29": {
+           |                "id": "$${json-unit.ignore}",
+           |                "blobId": "$${json-unit.ignore}",
```

Comment:
If we expose a blobId then the JMAP client should be able to download it. Which means "blobResolver".

But is it really needed?

IMO we could omit the blobId part of the JMAP model...